### PR TITLE
Add all fixes from 9X for custom modules

### PR DESCRIPTION
**Pull request by podarok**
Original Issue, this PR is going to fix: Get custom modules ready for 8.8+/9 Drupal core

## Steps for review

- [ ] Code review.
- [ ] Check Branch popups are working.
- [ ] Check Schedules.
- [ ] Verify OpenY AddThis module could be installed after uninstallation

Thank you for your contribution!

### CODE REVIEW

**Comment by gundevel** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2542/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2542/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2542/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2542/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2542/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2542/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2542/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2542/

File: modules/custom/openy_campaign/src/CampaignMenuService.php [hunk 1]
```diff
@@ -349,7 +349,7 @@ public function checkPermissions($node) {
   public function getActiveCampaigns() {
     // All dates are stored in database in UTC timezone.
     // Get current datetime in UTC timezone.
-    $dt = new \DateTime('now', new \DateTimezone(DATETIME_STORAGE_TIMEZONE));
+    $dt = new \DateTime('now', new \DateTimezone(DATETIME_ST\Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface::STORAGE_TIMEZONEORAGE_TIMEZONE));
```
Comment (AndreyMaximov) [inline modules/custom/openy_campaign/src/CampaignMenuService.php]: Typo

File: modules/custom/openy_analytics/src/AnalyticsCron.php [hunk 2]
```diff
@@ -276,7 +276,8 @@ function getParagraphsUsage() {
     $statement->addExpression('COUNT(type)', 'count');
     $paragraphs_counted = $statement->execute()->fetchAllKeyed();
 
-    $bundles = array_keys($this->entityTypeBundleInfo->getBundleInfo('paragraph'));
+    // $bundles = array_keys($this->entityTypeBundleInfo->getBundleInfo('paragraph'));
+    $bundles = array_keys(\Drupal::service('entity_type.bundle.info')->getBundleInfo('paragraph'));
```
Comment (anpolimus) [inline modules/custom/openy_analytics/src/AnalyticsCron.php]: We have to rebuild it according to DI.

File: modules/custom/openy_campaign/src/CampaignMenuService.php [hunk 3]
```diff
@@ -358,7 +358,7 @@ public function getActiveCampaigns() {
     $campaignIds = $nodeStorage->getQuery()
       ->condition('type', 'campaign')
       ->condition('status', TRUE)
-      ->condition('field_campaign_end_date', $now->format(DATETIME_DATETIME_STORAGE_FORMAT), '>=')
+      ->condition('field_campaign_end_date', $now->format(\Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface::DATETIME_STORAGE_FORMAT), '>=')
```
Comment (anpolimus) [inline modules/custom/openy_campaign/src/CampaignMenuService.php]: It looks like autogenerated, lets use USE at the top of the file

**Review by anpolimus — CHANGES_REQUESTED** [review]

File: modules/custom/openy_campaign/src/CampaignMenuService.php (hunk 1 above)
Comment (podarok) [reply modules/custom/openy_campaign/src/CampaignMenuService.php]: Thank you, fixed. Nice catch!

File: modules/custom/openy_analytics/src/AnalyticsCron.php (hunk 2 above)
Comment (podarok) [reply modules/custom/openy_analytics/src/AnalyticsCron.php]: This is code adoption, not a refactoring. All refactorings should happen after we are on Drupal 9

File: modules/custom/logger_entity/src/Entity/LoggerEntity.php [hunk 4]
```diff
@@ -150,7 +150,7 @@ public function isPublished() {
    * {@inheritdoc}
    */
   public function setPublished($published) {
-    $this->set('status', $published ? NODE_PUBLISHED : NODE_NOT_PUBLISHED);
+    $this->set('status', $published ? \Drupal\node\NodeInterface::PUBLISHED : \Drupal\node\NodeInterface::NOT_PUBLISHED);
```
Comment (hamrant) [inline modules/custom/logger_entity/src/Entity/LoggerEntity.php]: Namespaced classes/interfaces/traits should be referenced with use statements

File: modules/custom/openy_analytics/src/AnalyticsCron.php (hunk 2 above)
Comment (anpolimus) [reply modules/custom/openy_analytics/src/AnalyticsCron.php]: ok

**Comment by anpolimus** [comment]
As I said in chat, I'm worried about date functions changes. I'm not sure that the default PHP date functions, introduced here produce the same result at all use cases.
It is better to plan to do a regression testing at Activity Finder part of Open Y.

**Comment by AndreyMaximov** [comment]
@podarok Looks good overall, thank you.

@anpolimus the date functions are fine as per https://www.drupal.org/node/3009387

File: modules/custom/openy_addthis/openy_addthis.install [hunk 5]
```diff
@@ -10,7 +10,7 @@
  */
 function openy_addthis_install() {
   // Add our base field to the schema.
-  \Drupal::service('entity.definition_update_manager')->applyUpdates();
+  // TODO: \Drupal::service('entity.definition_update_manager')->applyUpdates();
```
Comment (hamrant) [inline modules/custom/openy_addthis/openy_addthis.install]: I think there should be code according to https://www.drupal.org/node/3034742

> Whenever an entity type or field storage definition needs to be created, changed or deleted, it has to be done with an explicit update function

File: modules/custom/openy_addthis/openy_addthis.install [hunk 6]
```diff
@@ -24,7 +24,7 @@ function openy_addthis_uninstall() {
     ->delete();
 
   // Remove our base field from the schema.
-  \Drupal::service('entity.definition_update_manager')->applyUpdates();
+  // TODO: \Drupal::service('entity.definition_update_manager')->applyUpdates();
```
Comment (hamrant) [inline modules/custom/openy_addthis/openy_addthis.install]: I think there should be code according to https://www.drupal.org/node/3034742

> Whenever an entity type or field storage definition needs to be created, changed or deleted, it has to be done with an explicit update function

File: modules/custom/openy_campaign/src/Entity/MemberCheckin.php [hunk 7]
```diff
@@ -166,8 +166,8 @@ public function postSave(EntityStorageInterface $storage, $update = TRUE) {
   private function getActiveCampaignMembers() {
     // Get the list of all active campaigns.
     $currentDate = new DrupalDateTime('now');
-    $currentDate->setTimezone(new \DateTimezone(DATETIME_STORAGE_TIMEZONE));
-    $formatted = $currentDate->format(DATETIME_DATETIME_STORAGE_FORMAT);
+    $currentDate->setTimezone(new \DateTimezone(\Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface::STORAGE_TIMEZONE));
```
Comment (hamrant) [inline modules/custom/openy_campaign/src/Entity/MemberCheckin.php]: Namespaced classes/interfaces/traits should be referenced with use statements

File: modules/custom/openy_campaign/src/Plugin/Block/CampaignRegisterBlock.php [hunk 8]
```diff
@@ -136,10 +136,10 @@ public function build() {
       $activeRegistration = FALSE;
     }
 
-    $utcCampaignStart = $localeCampaignStart->getDate()->format(DATETIME_DATETIME_STORAGE_FORMAT);
-    $utcCampaignEnd = $localeCampaignEnd->getDate()->format(DATETIME_DATETIME_STORAGE_FORMAT);
-    $utcCampaignRegStart = $localeRegistrationStart->getDate()->format(DATETIME_DATETIME_STORAGE_FORMAT);
-    $utcCampaignRegEnd = $localeRegistrationEnd->getDate()->format(DATETIME_DATETIME_STORAGE_FORMAT);
+    $utcCampaignStart = $localeCampaignStart->getDate()->format(\Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface::DATETIME_STORAGE_FORMAT);
```
Comment (hamrant) [inline modules/custom/openy_campaign/src/Plugin/Block/CampaignRegisterBlock.php]: Namespaced classes/interfaces/traits should be referenced with use statements

File: modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php [hunk 9]
```diff
@@ -116,7 +116,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
 
         \Drupal::messenger()->addWarning('There is manual crop set for this image. It overrides focal point settings');
         $status_messages = ['#type' => 'status_messages'];
-        $messages_html = drupal_render_root($status_messages);
+        $messages_html = \Drupal::service('renderer')->renderRoot($status_messages);
```
Comment (hamrant) [inline modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php]: Let's use DI here

File: modules/custom/openy_group_schedules/modules/groupex_form_cache/src/Entity/GroupexFormCache.php [hunk 10]
```diff
@@ -140,7 +140,7 @@ public function isPublished() {
    * {@inheritdoc}
    */
   public function setPublished($published) {
-    $this->set('status', $published ? NODE_PUBLISHED : NODE_NOT_PUBLISHED);
+    $this->set('status', $published ? \Drupal\node\NodeInterface::PUBLISHED : \Drupal\node\NodeInterface::NOT_PUBLISHED);
```
Comment (hamrant) [inline modules/custom/openy_group_schedules/modules/groupex_form_cache/src/Entity/GroupexFormCache.php]: Namespaced classes/interfaces/traits should be referenced with use statements

File: modules/custom/openy_mappings/src/Entity/Mapping.php [hunk 11]
```diff
@@ -149,7 +149,7 @@ public function isPublished() {
    * {@inheritdoc}
    */
   public function setPublished($published) {
-    $this->set('status', $published ? NODE_PUBLISHED : NODE_NOT_PUBLISHED);
+    $this->set('status', $published ? \Drupal\node\NodeInterface::PUBLISHED : \Drupal\node\NodeInterface::NOT_PUBLISHED);
```
Comment (hamrant) [inline modules/custom/openy_mappings/src/Entity/Mapping.php]: Namespaced classes/interfaces/traits should be referenced with use statements

File: modules/custom/openy_popups/src/Form/SettingsForm.php [hunk 12]
```diff
@@ -71,7 +71,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $config = \Drupal::service('config.factory')->getEditable('openy_popups.settings');
     if ($config->get('img')) {
       // Delete old image.
-      file_delete($config->get('img'));
+      \Drupal::service('file_system')->delete($config->get('img'));
```
Comment (hamrant) [inline modules/custom/openy_popups/src/Form/SettingsForm.php]: Let's use DI here

File: modules/custom/openy_socrates/modules/openy_autocomplete_path/openy_autocomplete_path.info.yml [hunk 13]
```diff
@@ -1,6 +1,6 @@
 name: Open Y Autocomplete Path
 type: module
 description: Implements an autocomplete matcher with path in the label.
-core: 8.x
+core_version_requirement: ^8 || ^9
 package: Open Y
 version: 8.x-1.0
```
Comment (hamrant) [inline modules/custom/openy_socrates/modules/openy_autocomplete_path/openy_autocomplete_path.info.yml]: No newline at end of file

**Review by hamrant — CHANGES_REQUESTED** [review]

**Comment by hamrant** [comment]
> This is code adoption, not a refactoring. All refactorings should happen after we are on Drupal 9

@podarok you can skip most of my comments according to this))

**Comment by gundevel** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2554/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2554/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2554/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2554/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2554/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2554/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2554/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2554/

File: modules/custom/openy_addthis/openy_addthis.install (hunk 5 above)
Comment (podarok) [reply modules/custom/openy_addthis/openy_addthis.install]: And we have no uncontrolled updates for storage definitions @hamrant 
So we can remove this code

**Comment by podarok** [comment]
retest this please

**Comment by gundevel** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2570/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2570/

**Comment by gundevel** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2570/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2570/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2570/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2570/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2570/blogWCAG2AAhtmlcs.txt

**Comment by gundevel** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2570/

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2644/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2644/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2644/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2644/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2644/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2644/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2644/blogWCAG2AAhtmlcs.txt

File: modules/custom/openy_analytics/src/AnalyticsCron.php (hunk 2 above)
Comment (podarok) [reply modules/custom/openy_analytics/src/AnalyticsCron.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_campaign/src/CampaignMenuService.php (hunk 3 above)
Comment (podarok) [reply modules/custom/openy_campaign/src/CampaignMenuService.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/logger_entity/src/Entity/LoggerEntity.php (hunk 4 above)
Comment (podarok) [reply modules/custom/logger_entity/src/Entity/LoggerEntity.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_campaign/src/Entity/MemberCheckin.php (hunk 7 above)
Comment (podarok) [reply modules/custom/openy_campaign/src/Entity/MemberCheckin.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_campaign/src/Plugin/Block/CampaignRegisterBlock.php (hunk 8 above)
Comment (podarok) [reply modules/custom/openy_campaign/src/Plugin/Block/CampaignRegisterBlock.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php (hunk 9 above)
Comment (podarok) [reply modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_group_schedules/modules/groupex_form_cache/src/Entity/GroupexFormCache.php (hunk 10 above)
Comment (podarok) [reply modules/custom/openy_group_schedules/modules/groupex_form_cache/src/Entity/GroupexFormCache.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_mappings/src/Entity/Mapping.php (hunk 11 above)
Comment (podarok) [reply modules/custom/openy_mappings/src/Entity/Mapping.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_popups/src/Form/SettingsForm.php (hunk 12 above)
Comment (podarok) [reply modules/custom/openy_popups/src/Form/SettingsForm.php]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

File: modules/custom/openy_socrates/modules/openy_autocomplete_path/openy_autocomplete_path.info.yml (hunk 13 above)
Comment (podarok) [reply modules/custom/openy_socrates/modules/openy_autocomplete_path/openy_autocomplete_path.info.yml]: fixed https://github.com/ymcatwincities/openy/pull/2174/commits/839562bfa412cf71c9862ca9d9ce54b72f92ff06

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2644/

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2645/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2645/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2645/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2645/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2645/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2645/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2645/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Open Y Upgrade Path site check installed at http://upgrade.openy.cibox.tools/build2645/

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2644/

File: modules/custom/openy_addthis/openy_addthis.install (hunk 6 above)
Comment (podarok) [reply modules/custom/openy_addthis/openy_addthis.install]: https://github.com/ymcatwincities/openy/pull/2174/commits/4e09e24c839250769eef5b4e73cf48b2c0331f01 fixed

File: modules/custom/openy_system/src/Form/OpenyModulesListForm.php [hunk 14]
```diff
@@ -91,7 +91,7 @@ protected function buildPackageRow(array $package) {
     $row['#requires'] = [];
     $row['#required_by'] = [];
 
-    $modules = system_rebuild_module_data();
+    $modules = \Drupal::service('extension.list.module')->reset()->getList();
```
Comment (hamrant) [inline modules/custom/openy_system/src/Form/OpenyModulesListForm.php:94]: Let's use DI here

File: modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php [hunk 15]
```diff
@@ -236,7 +243,7 @@ public static function ajaxSave(array $form, FormStateInterface $form_state) {
 
     \Drupal::messenger()->addStatus('Focal point updated');
     $status_messages = ['#type' => 'status_messages'];
-    $messages_html = drupal_render_root($status_messages);
+    $messages_html = \Drupal::service('renderer')->renderRoot($status_messages);
```
Comment (hamrant) [inline modules/custom/openy_focal_point/src/Form/OpenYFocalPointEditForm.php:246]: Please use `$this->renderer->renderRoot` we already have renderer on this form

**Review by hamrant — CHANGES_REQUESTED** [review]
> Namespaced classes/interfaces/traits should be referenced with use statements

In several places `\Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface::STORAGE_TIMEZONE` still not fixed

File: modules/custom/openy_system/src/Form/OpenyModulesListForm.php (hunk 14 above)
Comment (podarok) [reply modules/custom/openy_system/src/Form/OpenyModulesListForm.php:94]: Ok @hamrant . I'll work on this next week - form has no ContainerInterface implementation in order to pass DI

**Comment by podarok** [comment]
Outdated and mostly backported to date

**Pull request closed without merging**